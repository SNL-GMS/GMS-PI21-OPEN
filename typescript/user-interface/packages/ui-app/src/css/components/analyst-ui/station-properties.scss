.station-properties-display-window {
  height: 100%;
  width: 100%;
  padding: 0.5;
  display: flex;
  justify-content: center;
  align-items: center;
}

.station-properties-panel {
  display: flex;
  height: 100%;
  width: 100%;
  flex-direction: column;
  overflow-y: auto;
  overflow-x: hidden;
}

.station-properties-table {
  height: 100%;
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  // wrapper wraps a single table
  &__wrapper {
    height: 100%;
    flex-grow: 1;
    flex-basis: 0;
  }
}

.channel-configuration-table {
  padding-bottom: 0;
}
