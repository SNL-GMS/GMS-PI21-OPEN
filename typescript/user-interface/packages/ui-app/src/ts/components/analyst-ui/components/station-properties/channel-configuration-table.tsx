import { Classes, H4 } from '@blueprintjs/core';
import type { ChannelTypes, ResponseTypes, StationTypes } from '@gms/common-model';
import { delayExecutionReturnClearTimeout, formatTimeForDisplay } from '@gms/common-util';
import { Table } from '@gms/ui-core-components';
import classNames from 'classnames';
import React from 'react';

import {
  nonIdealStateLoadingStationDataQuery,
  nonIdealStateSelectChannelGroupRow
} from '~analyst-ui/components/station-properties/station-properties-non-ideal-states';
import {
  formatTimeShift,
  getChannelDataTypeForDisplay
} from '~analyst-ui/components/station-properties/station-properties-utils';
import { messageConfig } from '~analyst-ui/config/message-config';
import { defaultColumnDefinition } from '~common-ui/common/table-types';
import {
  formatNumberForDisplayFixedThreeDecimalPlaces,
  formatNumberForDisplayMaxThreeDecimalPlaces,
  getHeaderHeight,
  getRowHeightWithBorder,
  getTableCellStringValue
} from '~common-ui/common/table-utils';

import { ChannelColumnDefs } from './column-definitions';
import type {
  ChannelColumn,
  ChannelConfigurationRow,
  ChannelConfigurationTableProps
} from './types';

function parseChannelLocation(location: ChannelTypes.Location) {
  return {
    latitudeDegrees: location
      ? formatNumberForDisplayFixedThreeDecimalPlaces(location.latitudeDegrees)
      : messageConfig.invalidCellText,
    longitudeDegrees: location
      ? formatNumberForDisplayFixedThreeDecimalPlaces(location.longitudeDegrees)
      : messageConfig.invalidCellText,
    depthKm: location
      ? formatNumberForDisplayFixedThreeDecimalPlaces(location.depthKm)
      : messageConfig.invalidCellText,
    elevationKm: location
      ? formatNumberForDisplayFixedThreeDecimalPlaces(location.elevationKm)
      : messageConfig.invalidCellText
  };
}

function parseChannelResponse(response: ResponseTypes.Response) {
  return {
    calibrationFactor: response
      ? formatNumberForDisplayMaxThreeDecimalPlaces(response?.calibration?.calibrationFactor.value)
      : messageConfig.invalidCellText,
    calibrationPeriod: response
      ? formatNumberForDisplayMaxThreeDecimalPlaces(response?.calibration?.calibrationPeriodSec)
      : messageConfig.invalidCellText,
    calibrationStandardDeviation: response
      ? formatNumberForDisplayMaxThreeDecimalPlaces(
          response?.calibration?.calibrationFactor.standardDeviation
        )
      : messageConfig.invalidCellText,
    calibrationTimeShift: response
      ? formatTimeShift(response?.calibration?.calibrationTimeShift)
      : messageConfig.invalidCellText,
    calibrationEffectiveAt: response
      ? formatTimeForDisplay(response?.effectiveAt)
      : messageConfig.invalidCellText,
    calibrationResponseId: response
      ? getTableCellStringValue(response.id)
      : messageConfig.invalidCellText,
    fapResponseId: response
      ? getTableCellStringValue(response?.fapResponse?.id)
      : messageConfig.invalidCellText
  };
}

function formatChannelConfigurationTableRows(
  channels: ChannelTypes.Channel[],
  selectedStation: StationTypes.Station
): ChannelConfigurationRow[] {
  return channels.map(
    // eslint-disable-next-line complexity
    (chan: ChannelTypes.Channel, idx: number): ChannelConfigurationRow => {
      const relativePositions: Record<string, ChannelTypes.RelativePosition> =
        selectedStation.relativePositionsByChannel;
      const channelDisplacements = relativePositions[chan.name];
      const {
        calibrationEffectiveAt,
        calibrationFactor,
        calibrationPeriod,
        calibrationResponseId,
        calibrationStandardDeviation,
        calibrationTimeShift,
        fapResponseId
      } = parseChannelResponse(chan.response);
      const { latitudeDegrees, longitudeDegrees, depthKm, elevationKm } = parseChannelLocation(
        chan.location
      );
      return {
        id: `${idx}`,
        name: getTableCellStringValue(chan.name),
        channelBandType: getTableCellStringValue(chan.channelBandType),
        channelInstrumentType: getTableCellStringValue(chan.channelInstrumentType),
        channelOrientationType: getTableCellStringValue(chan.channelOrientationType),
        channelOrientationCode: getTableCellStringValue(chan.channelOrientationCode),
        channelDataType: getChannelDataTypeForDisplay(chan.channelDataType),
        nominalSampleRateHz: formatNumberForDisplayMaxThreeDecimalPlaces(chan.nominalSampleRateHz),
        description: chan.description
          ? getTableCellStringValue(chan.description.replace(/_/g, ' '))
          : messageConfig.invalidCellText,
        effectiveAt: formatTimeForDisplay(chan.effectiveAt),
        effectiveUntil: chan.effectiveUntil
          ? formatTimeForDisplay(chan.effectiveUntil)
          : messageConfig.latestVersionCellText,
        calibrationFactor,
        calibrationPeriod,
        calibrationStandardDeviation,
        calibrationTimeShift,
        calibrationEffectiveAt,
        calibrationResponseId,
        fapResponseId,
        orientationHorizontalDegrees: chan.orientationAngles
          ? formatNumberForDisplayMaxThreeDecimalPlaces(chan.orientationAngles.horizontalAngleDeg)
          : messageConfig.invalidCellText,
        orientationVerticalDegrees: chan.orientationAngles
          ? formatNumberForDisplayMaxThreeDecimalPlaces(chan.orientationAngles.verticalAngleDeg)
          : messageConfig.invalidCellText,
        latitudeDegrees,
        longitudeDegrees,
        depthKm,
        elevationKm,
        units: chan.units
          ? chan.units.slice(0, 1) + chan.units.slice(1).toLocaleLowerCase()
          : messageConfig.invalidCellText,
        northDisplacementKm: formatNumberForDisplayMaxThreeDecimalPlaces(
          channelDisplacements.northDisplacementKm
        ),
        verticalDisplacementKm: formatNumberForDisplayMaxThreeDecimalPlaces(
          channelDisplacements.verticalDisplacementKm
        ),
        eastDisplacementKm: formatNumberForDisplayMaxThreeDecimalPlaces(
          channelDisplacements.eastDisplacementKm
        )
      };
    }
  );
}

// eslint-disable-next-line react/function-component-definition
export const ChannelConfigurationTable: React.FunctionComponent<ChannelConfigurationTableProps> = (
  props: ChannelConfigurationTableProps
) => {
  const { channels, stationData, columnsToDisplay } = props;

  const tableRef = React.useRef<Table<ChannelConfigurationRow, unknown>>(null);

  const defaultColDefRef = React.useRef(defaultColumnDefinition<ChannelConfigurationRow>());

  const columnDefsRef = React.useRef(ChannelColumnDefs);
  React.useEffect(() => {
    return delayExecutionReturnClearTimeout(() => {
      if (tableRef && tableRef.current) {
        tableRef?.current?.updateVisibleColumns<ChannelColumn>(columnsToDisplay);
      }
    });
  }, [columnsToDisplay]);

  if (stationData && !channels) {
    return nonIdealStateSelectChannelGroupRow;
  }

  if (!stationData || !channels) {
    return nonIdealStateLoadingStationDataQuery;
  }
  const rowChannels: ChannelConfigurationRow[] = formatChannelConfigurationTableRows(
    channels,
    stationData
  );

  return (
    <div
      className={classNames(
        'channel-configuration-table',
        'ag-theme-dark',
        'station-properties-table',
        'with-separated-rows-color'
      )}
      data-cy="channel-configuration-table"
    >
      <H4 className={classNames(`${Classes.HEADING}`)}>Channel Configuration</H4>
      <div className={classNames(['station-properties-table__wrapper'])}>
        <Table<ChannelConfigurationRow, unknown>
          ref={ref => {
            tableRef.current = ref;
          }}
          context={{}}
          onGridReady={() => {
            if (tableRef && tableRef.current) {
              columnsToDisplay.forEach((shouldDisplay, columnName) => {
                tableRef.current?.setColumnVisible(columnName, shouldDisplay);
              });
            }
          }}
          defaultColDef={defaultColDefRef.current}
          columnDefs={columnDefsRef.current}
          rowData={rowChannels}
          rowHeight={getRowHeightWithBorder()}
          headerHeight={getHeaderHeight()}
          rowDeselection
          rowSelection="single"
          suppressCellFocus
          overlayNoRowsTemplate="No Channels to display"
          suppressContextMenu
          suppressDragLeaveHidesColumns
        />
      </div>
    </div>
  );
};
