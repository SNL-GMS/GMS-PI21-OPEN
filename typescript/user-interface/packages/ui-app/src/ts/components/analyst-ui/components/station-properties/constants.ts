export const headerCellBlockClass = 'station-properties-header-cell';
export const smallCellWidthPx = 105;
export const medCellWidthPx = 200;
export const largeCellWidthPx = 300;
