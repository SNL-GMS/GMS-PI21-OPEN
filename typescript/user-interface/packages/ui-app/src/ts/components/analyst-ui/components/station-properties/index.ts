import { StationPropertiesComponent } from './station-properties-component';
import { StationPropertiesPanel } from './station-properties-panel';

export { StationPropertiesComponent, StationPropertiesPanel };
