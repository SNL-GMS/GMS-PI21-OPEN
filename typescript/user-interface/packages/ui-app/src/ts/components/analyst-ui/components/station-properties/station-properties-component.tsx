import type { CommonTypes } from '@gms/common-model';
import { IanDisplays } from '@gms/common-model/lib/displays/types';
import type GoldenLayout from '@gms/golden-layout';
import { WithNonIdealStates } from '@gms/ui-core-components';
import {
  useAppSelector,
  useGetStationsEffectiveAtTimesQuery,
  useOperationalTimePeriodConfiguration
} from '@gms/ui-state';
import React from 'react';

import { BaseDisplay } from '~common-ui/components/base-display';

import {
  nonIdealStateEmptyEffectiveAtsQuery,
  nonIdealStateLoadingEffectiveAtsQuery,
  nonIdealStateNoOperationalTimePeriod,
  nonIdealStateSelectAStation,
  nonIdealStateTooManyStationsSelected
} from './station-properties-non-ideal-states';
import { StationPropertiesPanel } from './station-properties-panel';
import type { StationPropertiesComponentProps, StationPropertiesPanelProps } from './types';

type StationPropertiesPanelOrNonIdealStateProps = StationPropertiesPanelProps & {
  readonly glContainer?: GoldenLayout.Container;
  operationalTimeRange: CommonTypes.TimeRange;
  selectedStations: string[];
};

export const StationPropertiesPanelOrNonIdealState = WithNonIdealStates<
  StationPropertiesPanelOrNonIdealStateProps
>(
  [
    {
      condition: (props: StationPropertiesPanelOrNonIdealStateProps): boolean => {
        return props.selectedStations.length < 1;
      },
      element: nonIdealStateSelectAStation
    },
    {
      condition: (props: StationPropertiesPanelOrNonIdealStateProps): boolean => {
        return props.selectedStations.length > 1;
      },
      element: nonIdealStateTooManyStationsSelected
    },
    {
      condition: (props: StationPropertiesPanelOrNonIdealStateProps): boolean => {
        return !props.operationalTimeRange.startTimeSecs || !props.operationalTimeRange.endTimeSecs;
      },
      element: nonIdealStateNoOperationalTimePeriod
    },
    {
      condition: (props: StationPropertiesPanelOrNonIdealStateProps): boolean => {
        return !props.effectiveAtTimes;
      },
      element: nonIdealStateLoadingEffectiveAtsQuery
    },
    {
      condition: (props: StationPropertiesPanelOrNonIdealStateProps): boolean => {
        return props.effectiveAtTimes.length <= 0;
      },
      element: nonIdealStateEmptyEffectiveAtsQuery
    }
  ],
  StationPropertiesPanel
);

// eslint-disable-next-line react/function-component-definition
export const StationPropertiesComponent: React.FunctionComponent<StationPropertiesComponentProps> = (
  props: StationPropertiesComponentProps
) => {
  const { glContainer } = props;

  const unfilteredSelectedStations = useAppSelector(state => state.app.common?.selectedStationIds);
  const { timeRange } = useOperationalTimePeriodConfiguration();

  const operationalTimeRange: CommonTypes.TimeRange = {
    startTimeSecs: timeRange?.startTimeSecs,
    endTimeSecs: timeRange?.endTimeSecs
  };

  // TODO Instead of filtering on a string we need to update the selected station IDs to be objects with a type field
  const selectedStations = unfilteredSelectedStations.filter((item: string) => !item.includes('.'));
  const effectiveAtTimes: string[] = useGetStationsEffectiveAtTimesQuery({
    stationName: selectedStations[0],
    startTime: operationalTimeRange.startTimeSecs,
    endTime: operationalTimeRange.endTimeSecs
  }).data;

  return (
    <BaseDisplay
      glContainer={glContainer}
      className="station-properties-display-window"
      data-cy="station-properties-display-window"
      tabName={IanDisplays.STATION_PROPERTIES}
    >
      <StationPropertiesPanelOrNonIdealState
        selectedStation={selectedStations[0]}
        effectiveAtTimes={effectiveAtTimes}
        operationalTimeRange={operationalTimeRange}
        selectedStations={selectedStations}
      />
    </BaseDisplay>
  );
};
