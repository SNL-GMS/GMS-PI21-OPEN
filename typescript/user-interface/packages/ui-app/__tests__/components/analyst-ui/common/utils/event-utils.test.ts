import { ConfigurationTypes, EventTypes, LegacyEventTypes } from '@gms/common-model';
import type { LocationSolution } from '@gms/common-model/lib/event';
import type { Location } from '@gms/common-model/lib/station-definitions/channel-definitions/channel-definitions';
import type { EventStatus } from '@gms/ui-state';
import { signalDetectionsData } from '@gms/ui-state/__tests__/__data__';
import cloneDeep from 'lodash/cloneDeep';

import {
  determineDetectionColorLegacy,
  getDistanceToStationsForPreferredLocationSolutionId,
  getLocationToEventDistance,
  getPreferredHypothesisInStage,
  isAssociatedToCurrentEventHypothesisLegacy,
  isSignalDetectionAssociated,
  isSignalDetectionCompleteAssociated,
  isSignalDetectionOpenAssociated,
  isSignalDetectionOtherAssociated
} from '../../../../../src/ts/components/analyst-ui/common/utils/event-util';
import { eventData } from '../../../../__data__/event-data';
import { data } from '../../components/station-properties/mock-station-data';

const MOCK_DEGREES = 5;

const KM_TO_DEGREES = 111.1949266;
const MOCK_KM = KM_TO_DEGREES * MOCK_DEGREES;
const MOCK_BEARING = 10;

jest.mock('geolib', () => {
  return {
    getDistance: jest.fn(() => MOCK_KM * 1000),
    getGreatCircleBearing: jest.fn(() => MOCK_BEARING)
  };
});

describe('Event utils', () => {
  const mockLocationSolution: LocationSolution = {
    id: '',
    networkMagnitudeSolutions: [],
    featurePredictions: { featurePredictions: [] },
    locationBehaviors: [],
    location: {
      latitudeDegrees: 0,
      longitudeDegrees: 0,
      depthKm: 0,
      time: 0
    },
    locationRestraint: undefined,
    name: ''
  };
  it('getLocationToEventDistance returns a valid distance', () => {
    const { station } = data;

    expect(getLocationToEventDistance(station.location, mockLocationSolution.location)).toEqual({
      degrees: MOCK_DEGREES,
      km: MOCK_KM
    });
  });

  it('getDistanceToStationsForPreferredLocationSolutionId returns a valid distance array', () => {
    const stationLocation2: Location = {
      latitudeDegrees: 2,
      longitudeDegrees: 2,
      elevationKm: 2,
      depthKm: 0
    };

    const { station } = data;
    const station2 = cloneDeep(station);
    station2.location = stationLocation2;
    const sdData = cloneDeep(signalDetectionsData);
    sdData[0].station.name = 'STA';
    const associatedEvent = cloneDeep(eventData);
    // associate  the detection to the event
    associatedEvent.eventHypotheses[0].associatedSignalDetectionHypotheses.push(
      sdData[0].signalDetectionHypotheses[0]
    );

    expect(
      getDistanceToStationsForPreferredLocationSolutionId(
        associatedEvent,
        [station, station2],
        'AL1',
        sdData,
        station.allRawChannels.concat(station2.allRawChannels)
      )
    ).toEqual([
      { azimuth: MOCK_BEARING, distance: { degrees: MOCK_DEGREES, km: MOCK_KM }, id: 'STA' },
      {
        azimuth: MOCK_BEARING,
        distance: { degrees: MOCK_DEGREES, km: MOCK_KM },
        id: 'BPPPP.BPP01.CNN'
      },
      {
        azimuth: MOCK_BEARING,
        distance: { degrees: MOCK_DEGREES, km: MOCK_KM },
        id: 'BPPPP.BPP01.BBC'
      },
      { azimuth: MOCK_BEARING, distance: { degrees: MOCK_DEGREES, km: MOCK_KM }, id: 'STA' },
      {
        azimuth: MOCK_BEARING,
        distance: { degrees: MOCK_DEGREES, km: MOCK_KM },
        id: 'BPPPP.BPP01.CNN'
      },
      {
        azimuth: MOCK_BEARING,
        distance: { degrees: MOCK_DEGREES, km: MOCK_KM },
        id: 'BPPPP.BPP01.BBC'
      }
    ]);
  });

  it('determineDetectionColor for initial state', () => {
    const sd = signalDetectionsData[0];
    expect(determineDetectionColorLegacy(sd, undefined, 'foobar')).toEqual(
      ConfigurationTypes.defaultColorTheme.unassociatedSDColor
    );
    expect(determineDetectionColorLegacy(sd, [], 'foobar')).toEqual(
      ConfigurationTypes.defaultColorTheme.unassociatedSDColor
    );

    const event: LegacyEventTypes.Event = {
      id: 'legacyEvent',
      status: LegacyEventTypes.EventStatus.ReadyForRefinement,
      modified: true,
      hasConflict: false,
      currentEventHypothesis: undefined,
      conflictingSdIds: []
    };
    expect(determineDetectionColorLegacy(undefined, [event], 'foobar')).toEqual(
      ConfigurationTypes.defaultColorTheme.unassociatedSDColor
    );
    // TODO need to fix EventUtils to take new event type before testing more
    // of the functionality
    expect(determineDetectionColorLegacy(sd, [event], undefined)).toEqual(
      ConfigurationTypes.defaultColorTheme.unassociatedSDColor
    );
  });

  it('getPreferredHypothesisInStage returns a valid preferred event hypothesis', () => {
    expect(
      getPreferredHypothesisInStage(eventData.preferredEventHypothesisByStage, 'AL1')
    ).toMatchSnapshot();
  });

  it('getPreferredHypothesisInStage returns an undefined event hypothesis for BOGUS stage', () => {
    expect(
      getPreferredHypothesisInStage(eventData.preferredEventHypothesisByStage, 'BOGUS')
    ).toBeUndefined();
  });

  it('isSignalDetectionOpenAssociated detects signal detections  associated with the open event', () => {
    const associatedEvent = cloneDeep(eventData);
    const detection = cloneDeep(signalDetectionsData[0]);
    // associate  the detection to the event
    associatedEvent.eventHypotheses[0].associatedSignalDetectionHypotheses.push(
      detection.signalDetectionHypotheses[0]
    );
    // test with open event == associatedEvent
    expect(
      isSignalDetectionOpenAssociated(detection, [associatedEvent], associatedEvent.id)
    ).toBeTruthy();
    // test with open event != associatedEvent
    expect(
      isSignalDetectionOpenAssociated(detection, [associatedEvent], 'otherEventId')
    ).toBeFalsy();
  });

  it('isSignalDetectionCompleteAssociated detects signal detections  associated with a complete event', () => {
    const associatedEvent = cloneDeep(eventData);
    const detection = cloneDeep(signalDetectionsData[0]);
    const eventStatusesComplete: Record<string, EventStatus> = {
      '82ca9908-4272-4738-802b-f3d8f3099767': {
        stageId: { name: 'sample' },
        eventId: '82ca9908-4272-4738-802b-f3d8f3099767',
        eventStatusInfo: {
          eventStatus: EventTypes.EventStatus.COMPLETE,
          activeAnalystIds: ['user1', 'user2']
        }
      }
    };
    const eventStatusesInProg: Record<string, EventStatus> = {
      '82ca9908-4272-4738-802b-f3d8f3099767': {
        stageId: { name: 'sample' },
        eventId: '82ca9908-4272-4738-802b-f3d8f3099767',
        eventStatusInfo: {
          eventStatus: EventTypes.EventStatus.IN_PROGRESS,
          activeAnalystIds: ['user1', 'user2']
        }
      }
    };
    // associate  the detection to the event
    associatedEvent.eventHypotheses[0].associatedSignalDetectionHypotheses.push(
      detection.signalDetectionHypotheses[0]
    );
    // test with event status == complete
    expect(
      isSignalDetectionCompleteAssociated(detection, [associatedEvent], eventStatusesComplete)
    ).toBeTruthy();
    // test with open event != in progress
    expect(
      isSignalDetectionCompleteAssociated(detection, [associatedEvent], eventStatusesInProg)
    ).toBeFalsy();
  });

  it('isSignalDetectionOtherAssociated detects signal detections  associated with a non open event', () => {
    const associatedEvent = cloneDeep(eventData);
    const detection = cloneDeep(signalDetectionsData[0]);
    // associate  the detection to the event
    associatedEvent.eventHypotheses[0].associatedSignalDetectionHypotheses.push(
      detection.signalDetectionHypotheses[0]
    );
    // test with open event == associatedEvent
    expect(
      isSignalDetectionOtherAssociated(detection, [associatedEvent], associatedEvent.id)
    ).toBeFalsy();
    // test with open event != associatedEvent
    expect(
      isSignalDetectionOtherAssociated(detection, [associatedEvent], 'otherEventId')
    ).toBeTruthy();
  });

  it('isSignalDetectionAssociated detects signal detections  associated an event', () => {
    const associatedEvent = cloneDeep(eventData);
    const detection = cloneDeep(signalDetectionsData[0]);
    // associate  the detection to the event
    associatedEvent.eventHypotheses[0].associatedSignalDetectionHypotheses.push(
      detection.signalDetectionHypotheses[0]
    );
    // test with associated event
    expect(isSignalDetectionAssociated(detection, [associatedEvent])).toBeTruthy();
    // test with unassociated event
    expect(isSignalDetectionAssociated(detection, [eventData])).toBeFalsy();
  });
});

describe('Determine if a signal detection is associated to a legacy event', () => {
  const nullEvent = null;
  const undefinedEvent = undefined;
  const event: LegacyEventTypes.Event = {
    id: '1',
    status: LegacyEventTypes.EventStatus.AwaitingReview,
    modified: false,
    hasConflict: false,
    currentEventHypothesis: {
      processingStage: { id: '1' },
      eventHypothesis: {
        id: '1',
        rejected: false,
        event: {
          id: '1',
          status: LegacyEventTypes.EventStatus.AwaitingReview
        },
        associationsMaxArrivalTime: 1000000,
        signalDetectionAssociations: [
          {
            id: '1',
            rejected: false,
            signalDetectionHypothesis: {
              id: '20cc9505-efe3-3068-b7d5-59196f37992c',
              rejected: false,
              parentSignalDetectionId: '0'
            },
            eventHypothesisId: '1'
          }
        ],
        locationSolutionSets: [{ id: '1', count: 1, locationSolutions: [] }],
        preferredLocationSolution: { locationSolution: undefined }
      }
    },
    conflictingSdIds: []
  };
  const signalDetectionHypothesis = signalDetectionsData[0].signalDetectionHypotheses[0];
  test('Null event', () => {
    expect(
      isAssociatedToCurrentEventHypothesisLegacy(signalDetectionHypothesis, nullEvent)
    ).toEqual(false);
  });
  test('Undefined event', () => {
    expect(
      isAssociatedToCurrentEventHypothesisLegacy(signalDetectionHypothesis, undefinedEvent)
    ).toEqual(false);
  });
  test('Non-null event', () => {
    expect(isAssociatedToCurrentEventHypothesisLegacy(signalDetectionHypothesis, event)).toEqual(
      true
    );
  });
});
