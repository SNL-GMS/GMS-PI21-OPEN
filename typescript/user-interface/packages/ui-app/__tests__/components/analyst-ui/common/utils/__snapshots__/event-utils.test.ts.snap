// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event utils getPreferredHypothesisInStage returns a valid preferred event hypothesis 1`] = `
Object {
  "preferred": Object {
    "associatedSignalDetectionHypotheses": Array [],
    "id": Object {
      "eventId": "82ca9908-4272-4738-802b-f3d8f3099767",
      "hypothesisId": "44388ae5-5856-45e3-bf2f-4212a5af37c5",
    },
    "locationSolutions": Array [
      Object {
        "featurePredictions": Object {
          "featurePredictions": Array [],
        },
        "id": "c083f904-5083-4532-8b22-94eaf2426c7d",
        "location": Object {
          "depthKm": 1.1,
          "latitudeDegrees": 1.1,
          "longitudeDegrees": 1.1,
          "time": 0,
        },
        "locationBehaviors": Array [],
        "locationRestraint": Object {
          "depthRestraintType": "UNRESTRAINED",
          "positionRestraintType": "UNRESTRAINED",
          "timeRestraintType": "UNRESTRAINED",
        },
        "locationUncertainty": Object {
          "ellipses": Array [
            Object {
              "confidenceLevel": 0.5,
              "depthUncertaintyKm": 0,
              "kWeight": 0,
              "scalingFactorType": "CONFIDENCE",
              "semiMajorAxisLengthKm": 0,
              "semiMajorAxisTrendDeg": 0,
              "semiMinorAxisLengthKm": 0,
              "timeUncertainty": "PT5S",
            },
          ],
          "ellipsoids": Array [],
          "stdDevOneObservation": 1.1,
          "tt": 1.1,
          "xt": 1.1,
          "xx": 1.1,
          "xy": 1.1,
          "xz": 1.1,
          "yt": 1.1,
          "yy": 1.1,
          "yz": 1.1,
          "zt": 1.1,
          "zz": 1.1,
        },
        "name": "location solution one",
        "networkMagnitudeSolutions": Array [
          Object {
            "magnitude": Object {
              "standardDeviation": 0.1,
              "units": "MAGNITUDE",
              "value": 1.1,
            },
            "magnitudeBehaviors": Array [
              Object {
                "isDefining": true,
                "residual": 1.1,
                "stationMagnitudeSolution": Object {
                  "magnitude": Object {
                    "standardDeviation": 0.1,
                    "units": "MAGNITUDE",
                    "value": 1.1,
                  },
                  "measurement": Object {
                    "channel": Object {
                      "effectiveAt": 1636503404,
                      "name": "ASAR.AS01.SHZ",
                    },
                    "featureMeasurementType": "PHASE",
                    "measuredChannelSegment": Object {
                      "id": Object {
                        "channel": Object {
                          "effectiveAt": 1636503404,
                          "name": "ASAR.AS01.SHZ",
                        },
                        "creationTime": 1636503404,
                        "endTime": 1636503504,
                        "startTime": 1636503404,
                      },
                    },
                    "measurementValue": Object {
                      "confidence": 0.5,
                      "value": "P",
                    },
                    "snr": null,
                  },
                  "model": "UNKNOWN",
                  "phase": "UNKNOWN",
                  "station": Object {
                    "allRawChannels": Array [
                      Object {
                        "canonicalName": "Canonical Name One",
                        "channelBandType": "B",
                        "channelDataType": "DIAGNOSTIC_SOH",
                        "channelInstrumentType": "H",
                        "channelOrientationCode": "Z",
                        "channelOrientationType": "Z",
                        "configuredInputs": Array [],
                        "description": "Example description",
                        "effectiveAt": 0,
                        "effectiveUntil": 1634680114.58,
                        "location": Object {
                          "depthKm": 100,
                          "elevationKm": 5500,
                          "latitudeDegrees": 35,
                          "longitudeDegrees": -125,
                        },
                        "name": "Real Channel Name One",
                        "nominalSampleRateHz": 65,
                        "orientationAngles": Object {
                          "horizontalAngleDeg": 65,
                          "verticalAngleDeg": 135,
                        },
                        "processingDefinition": undefined,
                        "processingMetadata": undefined,
                        "response": Object {
                          "calibration": Object {
                            "calibrationFactor": Object {
                              "standardDeviation": 0.112,
                              "units": "SECONDS",
                              "value": 1.2,
                            },
                            "calibrationPeriodSec": 14.5,
                            "calibrationTimeShift": "PT2S",
                          },
                          "effectiveAt": 0,
                          "effectiveUntil": 253370764800,
                          "fapResponse": undefined,
                          "id": "5dc3dce4-ae3c-3fb0-84ba-2f83fb5a89d2",
                        },
                        "station": Object {
                          "name": "STA",
                        },
                        "units": "NANOMETERS",
                      },
                      Object {
                        "canonicalName": "Canonical Name Two",
                        "channelBandType": "B",
                        "channelDataType": "DIAGNOSTIC_SOH",
                        "channelInstrumentType": "H",
                        "channelOrientationCode": "Z",
                        "channelOrientationType": "Z",
                        "configuredInputs": Array [],
                        "description": "Example description",
                        "effectiveAt": 0,
                        "effectiveUntil": 1634680114.582,
                        "location": Object {
                          "depthKm": 100,
                          "elevationKm": 5500,
                          "latitudeDegrees": 35,
                          "longitudeDegrees": -125,
                        },
                        "name": "Real Channel Name Two",
                        "nominalSampleRateHz": 65,
                        "orientationAngles": Object {
                          "horizontalAngleDeg": 65,
                          "verticalAngleDeg": 135,
                        },
                        "processingDefinition": undefined,
                        "processingMetadata": undefined,
                        "response": Object {
                          "calibration": Object {
                            "calibrationFactor": Object {
                              "standardDeviation": 0.112,
                              "units": "SECONDS",
                              "value": 1.2,
                            },
                            "calibrationPeriodSec": 14.5,
                            "calibrationTimeShift": "PT2S",
                          },
                          "effectiveAt": 0,
                          "effectiveUntil": 253370764800,
                          "fapResponse": undefined,
                          "id": "a9497b80-f49c-3bdc-9283-7c64b5562cd6",
                        },
                        "station": Object {
                          "name": "STA",
                        },
                        "units": "NANOMETERS",
                      },
                    ],
                    "channelGroups": Array [
                      Object {
                        "channels": Array [
                          Object {
                            "canonicalName": "Canonical Name One",
                            "channelBandType": "B",
                            "channelDataType": "DIAGNOSTIC_SOH",
                            "channelInstrumentType": "H",
                            "channelOrientationCode": "Z",
                            "channelOrientationType": "Z",
                            "configuredInputs": Array [],
                            "description": "Example description",
                            "effectiveAt": 0,
                            "effectiveUntil": 1634680114.58,
                            "location": Object {
                              "depthKm": 100,
                              "elevationKm": 5500,
                              "latitudeDegrees": 35,
                              "longitudeDegrees": -125,
                            },
                            "name": "Real Channel Name One",
                            "nominalSampleRateHz": 65,
                            "orientationAngles": Object {
                              "horizontalAngleDeg": 65,
                              "verticalAngleDeg": 135,
                            },
                            "processingDefinition": undefined,
                            "processingMetadata": undefined,
                            "response": Object {
                              "calibration": Object {
                                "calibrationFactor": Object {
                                  "standardDeviation": 0.112,
                                  "units": "SECONDS",
                                  "value": 1.2,
                                },
                                "calibrationPeriodSec": 14.5,
                                "calibrationTimeShift": "PT2S",
                              },
                              "effectiveAt": 0,
                              "effectiveUntil": 253370764800,
                              "fapResponse": undefined,
                              "id": "5dc3dce4-ae3c-3fb0-84ba-2f83fb5a89d2",
                            },
                            "station": Object {
                              "name": "STA",
                            },
                            "units": "NANOMETERS",
                          },
                        ],
                        "description": "Channel Group Description",
                        "effectiveAt": 0,
                        "effectiveUntil": 253370764800,
                        "location": Object {
                          "depthKm": 100,
                          "elevationKm": 5500,
                          "latitudeDegrees": 35,
                          "longitudeDegrees": -125,
                        },
                        "name": "Test Channel Group",
                        "type": "PROCESSING_GROUP",
                      },
                    ],
                    "description": "This is a test station",
                    "effectiveAt": 1634678314.635,
                    "effectiveUntil": 253370764800,
                    "location": Object {
                      "depthKm": 50,
                      "elevationKm": 10,
                      "latitudeDegrees": 35.647,
                      "longitudeDegrees": 100,
                    },
                    "name": "STA",
                    "relativePositionsByChannel": Object {
                      "Real Channel Name One": Object {
                        "eastDisplacementKm": 5,
                        "northDisplacementKm": 50,
                        "verticalDisplacementKm": 10,
                      },
                      "Real Channel Name Two": Object {
                        "eastDisplacementKm": 5,
                        "northDisplacementKm": 50,
                        "verticalDisplacementKm": 10,
                      },
                    },
                    "type": "HYDROACOUSTIC",
                  },
                  "type": "MB",
                },
                "weight": 1.1,
              },
            ],
            "type": "MB",
          },
        ],
      },
    ],
    "name": "preferred one",
    "parentEventHypotheses": Array [],
    "preferredLocationSolution": Object {
      "id": "c083f904-5083-4532-8b22-94eaf2426c7d",
    },
    "rejected": false,
  },
  "preferredBy": "ANALYST1",
  "stage": Object {
    "name": "AL1",
  },
}
`;
